### DISCUSSION TITLE
Avoiding "ESLint: Unsafe assignment of an `any` value" in GetStaticProps

### DISCUSSION THREAD

**Discussion by cesalberca** [amannn/next-intl#445] [Q&A]
I followed the documentation and everything is working correctly. However I have an eslint error regarding an unsafe assignment of any value. I'm guessing I'm missing a cast, but I don't really know what to cast the await import(\`../../public/locales/${locale}.json\`) call to. Here is the code:

```
export async function getStaticProps({ locale }: GetStaticPropsContext) {
  return {
    props: {
      messages: (await import(`../../public/locales/${locale}.json`)).default,
    },
  }
}
```

This is the eslint error message:

```
ESLint: Unsafe assignment of an `any` value.(@typescript-eslint/no-unsafe-assignment)
```

What would be the type?

Thanks!

**Comment by hb20007** [comment]
I have the same issue in my `i18n.ts` file, which looks like this:
```ts
import { getRequestConfig } from 'next-intl/server';

export default getRequestConfig(async ({ locale }) => ({
  messages: (await import(`../messages/${locale}.json`)).default,
}));
```

  4:3   error  Unsafe assignment of an `any` value                        @typescript-eslint/no-unsafe-assignment
  4:58  error  Unsafe member access .default on an `any` value  @typescript-eslint/no-unsafe-member-access

---

Also tried to update the file to match the [current documentation](https://next-intl-docs.vercel.app/docs/getting-started/app-router#i18nts) exactly, but got even more errors:

```ts
import localeInfo from '@/localeInfo';
import { getRequestConfig } from 'next-intl/server';
import { notFound } from 'next/navigation';

const { locales } = localeInfo;

export default getRequestConfig(async ({ locale }) => {
  // Validate that the incoming `locale` parameter is valid
  if (!locales.includes(locale as any)) notFound();

  return {
    messages: (await import(`../messages/${locale}.json`)).default,
  };
});
```

   9:25  error  Unsafe argument of type `any` assigned to a parameter of type `string`  @typescript-eslint/no-unsafe-argument
   9:35  error  Unexpected any. Specify a different type                                @typescript-eslint/no-explicit-any
  12:5   error  Unsafe assignment of an `any` value                                     @typescript-eslint/no-unsafe-assignment
  12:60  error  Unsafe member access .default on an `any` value                         @typescript-eslint/no-unsafe-member-access

**Reply by hb20007** [reply]
@cesalberca See https://github.com/amannn/next-intl/issues/991#issuecomment-2050087509 for comments from the creator of `next-intl` regarding this.
